### PR TITLE
Make GPS coord listing always display the activated GPS first.

### PR DESCRIPTION
**Pull request by Allspice**
Instead of e.g. the coords for your held GPS being buried under 20+ GPS printed for vaulthunting.

:cl:
* tweak: The GPS you are looking at will always display its coords at the top of the coords list.

### CODE REVIEW

File: nano/templates/gps.tmpl [hunk 1]
```diff
@@ -24,11 +24,15 @@ Used In File(s): \code\modules\telesci\gps.dm
 </div>
 
 <h3>Detected signals</h3>
+<div class="item">
+	<div class="itemLabelNarrow">{{:data.gpstag}}</div>
+	<div class="itemContentWide">{{:data.location_text}}</div>
+</div>
 {{for data.devices}}
 	<div class="item">
 		<div class="itemLabelNarrow">{{:value.tag}}</div>
```
Comment (Kurfursten) [inline nano/templates/gps.tmpl:33]: If nano has `continue` you could make it skip your own gps on the main list, to reduce bloat.

File: code/modules/telesci/gps.dm [hunk 2]
```diff
@@ -16,6 +16,7 @@ var/list/SPS_list = list()
 	var/emped = FALSE
 	var/autorefreshing = FALSE
 	var/builtin = FALSE
+	var/unique_id = ""
```
Comment (DamianX) [inline code/modules/telesci/gps.dm]: Remove this, just use \ref[] when\if needed

File: code/modules/telesci/gps.dm [hunk 3]
```diff
@@ -61,33 +63,39 @@ var/list/SPS_list = list()
...
+
 /obj/item/device/gps/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open=NANOUI_FOCUS)
 	var/data[0]
 	if(emped)
 		data["emped"] = TRUE
 	else
 		data["gpstag"] = gpstag
 		data["autorefresh"] = autorefreshing
+		data["location_text"] = get_location_name()
+		data["unique_id"] = unique_id
```
Comment (DamianX) [inline code/modules/telesci/gps.dm]: You're never using this in the template, why is this here?

File: code/modules/telesci/gps.dm [hunk 4]
```diff
@@ -61,33 +63,39 @@ var/list/SPS_list = list()
...
-			if(device_rip)
-				device_data["location_text"] = "ERROR"
-			else if(!device_turf || !device_area)
-				device_data["location_text"] = "UNKNOWN"
-			else if(device_turf.z > WORLD_X_OFFSET.len)
-				device_data["location_text"] = "[format_text(device_area.name)] (UNKNOWN, UNKNOWN, UNKNOWN)"
-			else
-				device_data["location_text"] = "[format_text(device_area.name)] ([device_turf.x-WORLD_X_OFFSET[device_turf.z]], [device_turf.y-WORLD_Y_OFFSET[device_turf.z]], [device_turf.z])"
-			devices += list(device_data)
+			if(src.unique_id != G.unique_id)
```
Comment (DamianX) [inline code/modules/telesci/gps.dm]: You can just compare `src` and `G` directly

File: code/modules/telesci/gps.dm [hunk 5]
```diff
@@ -61,33 +63,39 @@ var/list/SPS_list = list()
...
+				var/device_data[0]
+				var/device_tag = null
```
Comment (DamianX) [inline code/modules/telesci/gps.dm]: What is this? Just replace thee three lines with `device_data["tag"] = G.gpstag`, no?

File: code/modules/telesci/gps.dm [hunk 6]
```diff
@@ -61,33 +63,39 @@ var/list/SPS_list = list()
...
+				device_tag = G.gpstag
+				device_data["tag"] = device_tag
+				device_data["location_text"] = G.get_location_name()
+				device_data["unique_id"] = G.unique_id
```
Comment (DamianX) [inline code/modules/telesci/gps.dm]: delet

File: code/modules/telesci/gps.dm (hunk 3 above)
Comment (Allspice) [reply code/modules/telesci/gps.dm]: location_text is used https://github.com/vgstation-coders/vgstation13/pull/18242#discussion-diff-187188479R27

Comment (DamianX) [reply code/modules/telesci/gps.dm]: Yes, I was referring to unique_id

File: nano/templates/gps.tmpl (hunk 1 above)
Comment (DamianX) [reply nano/templates/gps.tmpl:33]: This is done already in DM

**Comment by DamianX** [comment]
Code looks good, can we see a screenshot?

**Comment by Allspice** [comment]
GPS Image
